### ISSUE TITLE
resendSignUpCode doesn't work only on iOS. 

### ISSUE THREAD

**Issue by anush2593** [aws-amplify/amplify-swift#2791]
### Describe the bug

After the user signs in successfully, we have an option that lets the user to request a code again.
Every other api call works besides this one

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Sign the user
2. Get the confirmation that the user signed in
3. Navigate to the screen where the user can fill the confirmation code that comes to the email
4. Press resend confirmation code button with calls resendSignUpCode 
5. I get the email with code, but I get an unexpected error
```

### Expected behavior

If the sign up was successful I would expect to get a success message

### Amplify Framework Version

2.0.1

### Amplify Categories

API, Auth

### Dependency manager

Swift PM

### Swift version

5.5.1

### CLI version

6.3.1

### Xcode version

14.0.1 (14A400)

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


INSERT LOG MESSAGES HERE
```
</details>
```


### Is this a regression?

Yes

### Regression additional context

_No response_

### Device

iPhone mini SE

### iOS Version

iOS 16.1

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by atierian** [comment]
Hi @anush2593, thanks for opening this issue. 

The issue you're experiencing was fixed and released in [2.0.2](https://github.com/aws-amplify/amplify-swift/releases/tag/v2.0.2). Upgrading to that version (or any version after it) will resolve this.

Thanks!

**Comment by harsh62** [comment]
Duplicate of #2792 .
